nri-env-parser/src/Environment.hs

```diff
@@ -234,13 +235,27 @@ instance Applicative Decoder where
           (fr, xr) ->
             fr <*> xr
 
+instance Monad Decoder where
+  (Decoder consumes f) >>= toDecoder2 =
+    Decoder
+      { consumes = consumes,
+        readFromEnvironment = \env ->
+          andThen
+            ( \val ->
+                toDecoder2 val
+                  |> readFromEnvironment
+                  |> \f' -> f' env
+            )
+            (f env)
+      }
+
```

Comment:
it's unfortunate that this isn't compatible with the `Applicative` instance semantics re: capturing both errors but there's really no choice there

Reply:
The second decoder only comes into existence if the first once succeeded, as opposed to the applicative semantics which can happen in parallel; so yeah, no choice indeed!